app/models/host/managed.rb

```diff
@@ -539,7 +539,7 @@ def jumpstart?
 
   def set_hostgroup_defaults
     return unless hostgroup
-    assign_hostgroup_attributes(%w{environment_id domain_id compute_profile_id})
```

Comment:
Removing `environment_id` here is simply making the Host not inherit it at all from Host group. 
I think you're doing it because otherwise the `before_validation` will set the environment_id, but so it does with other non-critical items (`compute_profile_id`). 

The biggest issue is that we'd not inherit the `environment` in any case with this change, so I'm afraid you'll have to look for a different way to not 'auto-inherit' the attribute on save.